invoice_sale_delivery_date/tests/test_invoicing.py

```diff
@@ -0,0 +1,122 @@
+from datetime import timedelta
+
+from odoo import fields
+from odoo.addons.stock_account.tests.test_anglo_saxon_valuation_reconciliation_common import \
+    ValuationReconciliationTestCommon
+from odoo.tests import tagged, Form
+
+
+@tagged('post_install', '-at_install')
+class TestSaleExpectedDate(ValuationReconciliationTestCommon):
+
+    def _do_invoicing(self, sale: 'sale.order') -> 'account.move':
+        """
+        Invoice the given sale
+        """
+        context = {
+            "active_model": sale._name,
+            "active_id": sale.id,
+            "active_ids": sale.ids,
+            "open_invoices": True,  # To get the invoices back
+        }
+        InvoiceWizard = \
+            self.env["sale.advance.payment.inv"].with_context(**context)
+        wizard: InvoiceWizard = Form(InvoiceWizard)
```

Comment:
Cool! I'm not used to seeing type hints in our code, so it threw me off at first :smile:.